src/util/serializing/ObjectInputStream.cpp

```diff
@@ -94,94 +76,46 @@ auto ObjectInputStream::readString() -> std::string {
-        throw InputStreamException("End reached, but try to read an string", __FILE__, __LINE__);
-    }
-
     std::string output;
     output.resize(lenString);
 
     istream.read(&output[0], (long)lenString);
-
-    return output;
-}
-
-void ObjectInputStream::readData(void** data, int* length) {
-    checkType('b');
-
-    if (istream.str().size() < 2 * sizeof(int)) {
-        throw InputStreamException("End reached, but try to read data len and width", __FILE__, __LINE__);
-    }
-
-    int len = readTypeFromSStream<int>(istream);
-    int width = readTypeFromSStream<int>(istream);
-
-    if (istream.str().size() < static_cast<size_t>(len * width)) {
-        throw InputStreamException("End reached, but try to read data", __FILE__, __LINE__);
-    }
-
-    if (len == 0) {
-        *length = 0;
-        *data = nullptr;
-    } else {
-        *data = (void*)new char[(size_t)(len * width)];
-        *length = len;
-
-        istream.read((char*)*data, len * width);
-    }
-}
-
-template <typename T>
-void ObjectInputStream::readData(std::vector<T>& data) {
-    checkType('b');
-
-    if (istream.str().size() < 2 * sizeof(int)) {
-        throw InputStreamException("End reached, but try to read data len and width", __FILE__, __LINE__);
-    }
-
-    int len = readTypeFromSStream<int>(istream);
-    int width = readTypeFromSStream<int>(istream);
-
-    if (width != sizeof(T)) {
-        throw InputStreamException("Data width mismatch requested type width", __FILE__, __LINE__);
+    if (istream.fail()) {
+        throw InputStreamException("End reached, but try to read an string", __FILE__, __LINE__);
     }
 
-    if (istream.str().size() < static_cast<size_t>(len * width)) {
-        throw InputStreamException("End reached, but try to read data", __FILE__, __LINE__);
-    }
 
-    if (len) {
-        data.resize(len);
-        istream.read((char*)data.data(), len * width);
-    }
+    return output;
 }
 
-auto ObjectInputStream::readImage() -> std::string {
+auto ObjectInputStream::readImage() -> const std::vector<std::byte> {
     checkType('m');
 
-    if (istream.str().size() < sizeof(size_t)) {
+
+    const size_t len = readTypeFromSStream<size_t>(istream);
+    if (istream.fail()) {
         throw InputStreamException("End reached, but try to read an image's data's length", __FILE__, __LINE__);
     }
 
-    const size_t len = readTypeFromSStream<size_t>(istream);
-    if (istream.str().size() < len) {
+    std::vector<std::byte> data;
+    data.resize(len);
+    istream.read(reinterpret_cast<char*>(data.data()), static_cast<int>(len));
+    if (istream.fail()) {
         throw InputStreamException("End reached, but try to read an image", __FILE__, __LINE__);
     }
-    std::string data;
-    data.resize(len);
-    istream.read(data.data(), static_cast<int>(len));
 
     return data;
 }
 
 void ObjectInputStream::checkType(char type) {
-    if (istream.str().size() < 2) {
+    char t = 0, underscore = 0;
+    istream >> underscore >> t;
```

Comment:
**Question:** is the istream binary or formatted? Else, `>>` should not be used.

Reply:
The stream is binary. I have changed it to use `istream::read`